Training/python/2018v1/Training_v0p1.py

```diff
@@ -480,7 +482,12 @@ def main(cfg: DictConfig) -> None:
         run_kwargs = {'experiment_id': experiment_id}
 
     # run the training with mlflow tracking
-    with mlflow.start_run(**run_kwargs) as active_run:
+    with mlflow.start_run(**run_kwargs) as main_run:
```

Comment:
Unless I am mistaken, with the current mlflow setup still a new parent run will be created and then a child run nested inside of it. While the original idea was that nested runs (corresponding to fine-tuning of a pretrained model), will be placed inside of an already existing run (corresponding to  the pretrained model).

This is because `run_kwargs` doesn't contain information about pretrained run to be started first as a parent run. So the corresponding case should be added to `run_kwargs` definition, for example like:

```python
 experiment = mlflow.get_experiment_by_name(cfg.experiment_name) 
 if experiment is not None: # fetch existing experiment id 
     run_kwargs = {'experiment_id': experiment.experiment_id}
     if cfg["pretrained"] is not None: # initialise with pretrained run, otherwise skip and create a new run
         run_kwargs['run_id'] = cfg["pretrained"]["run_id"]
 else: # create new experiment 
     experiment_id = mlflow.create_experiment(cfg.experiment_name) 
     run_kwargs = {'experiment_id': experiment_id} 

with mlflow.start_run(**run_kwargs) as main_run:
### ...
```